<?xml version="1.0"?>
<launch>
    
    <arg name="n" default="0"/>
    <arg name="radius" default=".15"/>
    <arg name="use_move_base" default="false"/>
    <arg name="zero_pose" default="false"/>
    
    <arg name="robot" value="costmap"/>
    
    <!-- purge logs -->
    <param name="purged" command="rosclean purge -y"/>
    
   <node name="mrs_monitor" pkg="mrs_monitor" type="mrs_monitor" respawn="true" output="screen"/>
   
    <!-- A single global planner for all robots -->
   <node name="global_costmap" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 map global_costmap 100"/>
   <node name="mrs_planner" pkg="mrs_monitor" type="mrs_planner" respawn="true">
        <rosparam file="$(find mrs_monitor)/param/costmap_common.yaml" command="load" ns="global_costmap"  subst_value="true"/>
        <rosparam file="$(find mrs_monitor)/param/global_costmap.yaml" command="load"  subst_value="true"/>
        <rosparam file="$(find mrs_monitor)/param/planner.yaml" command="load" subst_value="true"/>
   </node>   
   
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 0)">
        <arg name="robot" value="robot1"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="255"/>
        <arg name="G" value="0"/>
        <arg name="B" value="0"/>
        <arg name="x" value="0"/>
        <arg name="y" value="0"/>
        <arg name="theta" value="0"/>     
        <arg name="radius" value="$(arg radius)"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 1)">
        <arg name="robot" value="robot2"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="0"/>
        <arg name="G" value="255"/>
        <arg name="B" value="0"/>
        <arg name="x" value="4"/>
        <arg name="y" value="2.3"/>
        <arg name="theta" value="1.5"/>     
        <arg name="radius" value="$(arg radius)"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 2)">
        <arg name="robot" value="robot3"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="0"/>
        <arg name="G" value="0"/>
        <arg name="B" value="255"/>
        <arg name="x" value="-3.3"/>
        <arg name="y" value="-3.3"/>
        <arg name="radius" value="$(arg radius)"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 3)">
        <arg name="robot" value="robot4"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="255"/>
        <arg name="G" value="255"/>
        <arg name="B" value="0"/>
        <arg name="x" value="8.7"/>
        <arg name="y" value="-3.3"/>
        <arg name="theta" value="-1.5"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 4)">
        <arg name="robot" value="robot5"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="255"/>
        <arg name="G" value="0"/>
        <arg name="B" value="255"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>                
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 5)">
        <arg name="robot" value="robot6"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="0"/>
        <arg name="G" value="255"/>
        <arg name="B" value="255"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 6)">
        <arg name="robot" value="robot7"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="100"/>
        <arg name="G" value="200"/>
        <arg name="B" value="100"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 7)">
        <arg name="robot" value="robot8"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="200"/>
        <arg name="G" value="100"/>
        <arg name="B" value="100"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 8)">
        <arg name="robot" value="robot9"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="100"/>
        <arg name="G" value="100"/>
        <arg name="B" value="200"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
    <include file="$(find mrs_monitor)/launch/spawn.launch" if="$(eval arg('n') > 9)">
        <arg name="robot" value="robot10"/>
        <arg name="zero_pose" value="$(arg zero_pose)"/>
        <arg name="R" value="100"/>
        <arg name="G" value="100"/>
        <arg name="B" value="100"/>
        <arg name="use_move_base" value="$(arg use_move_base)"/>        
    </include>
    
</launch>
